File: puzzles/python3/conway-sequence/conway_sequence.py
from typing import List


if __name__ == "__main__":
    original = int(input())
    line_number = int(input())
    conway_sequence: List[int] = [original]

    for line in range(1, line_number):
        temp_sequence: List[int] = []
        count = 0
        previous = conway_sequence[0]
        for number in conway_sequence:
            if number == previous:
                count += 1
            else:
                temp_sequence.append(count)
                temp_sequence.append(previous)
                previous = number
                count = 1

        temp_sequence.append(count)
        temp_sequence.append(previous)
        conway_sequence = temp_sequence  # update

    print(' '.join(map(str, conway_sequence)))

File: puzzles/cs/conway-sequence/conway-sequence.cs
using System;
using System.Collections.Generic;

public class Program
{
    public static void Main()
    {
        int original = int.Parse(Console.ReadLine());
        int lineNumber = int.Parse(Console.ReadLine());
        List<int> conwaySequence = new List<int> { original };

        for (int line = 1; line < lineNumber; line++)
        {
            List<int> tempSequence = new List<int>();
            int count = 0;
            int previous = conwaySequence[0];
            foreach (int number in conwaySequence)
            {
                if (number == previous)
                {
                    count += 1;
                }
                else
                {
                    tempSequence.Add(count);
                    tempSequence.Add(previous);
                    previous = number;
                    count = 1;
                }
            }

            tempSequence.Add(count);
            tempSequence.Add(previous);
            conwaySequence = tempSequence;  // update
        }

        Console.WriteLine(string.Join(" ", conwaySequence));
    }
}
